warcry_proj/warcry_proj/views/default.py

```diff
@@ -0,0 +1,139 @@
+
+    def dict_to_object(self, d):
+        if '__type__' not in d:
+            return d
+
+        type = d.pop('__type__')
+        try:
+            dateobj = datetime(**d)
+            return dateobj
+        except:
+            d['__type__'] = type
+            return d
+
+class DateTimeEncoder(JSONEncoder):
+    """ Instead of letting the default encoder convert datetime to string,
+        convert datetime objects into a dict, which can be decoded by the
+        DateTimeDecoder
+    """
+
+    def default(self, obj):
+        if isinstance(obj, datetime):
+            return {
+                '__type__' : 'datetime',
+                'year' : obj.year,
+                'month' : obj.month,
+                'day' : obj.day,
+                'hour' : obj.hour,
+                'minute' : obj.minute,
+                'second' : obj.second,
+                'microsecond' : obj.microsecond,
+            }
+        else:
+            return JSONEncoder.default(self, obj)
+
+#Get a list of all Users in Database
+@view_config(route_name='all_users', request_method='GET', renderer='json')
+def get_all_users(_):
+    schema = UserSchema(many=True)
+    return schema.dumps(DBSession.query(User).all(), cls=DateTimeEncoder)
+
+#Get a single user by specifying the user ID
+@view_config(route_name='single_user', request_method='GET', renderer='json')
+def get_single_user(request):
+    user_id = request.matchdict.get('user_id')
+    if user_id == '__first__':
+        user_id = UserSchema().dumps(DBSession.query(User).get(1), cls=DateTimeEncoder)
+        user_id = UserSchema().dumps(DBSession.query(User).get(1)).id
+
+    user = UserSchema().dumps(DBSession.query(User).get(user_id), cls=DateTimeEncoder)
+    if not user:
+        msg = "The User with id '{}' was not found.".format(user_id)
+        return Response(status=404, json_body={'error': msg})
+
+    return user
+
+#Create / Insert a User
+@view_config(route_name='create_new_user', request_method='POST', renderer='json')
+def create_user(request):
+     data = request.json_body
+     schema = UserSchema()
+     json_dict = schema.load(data).data
+     entered_name = json_dict["name"]
+     #Super Hero
+     superHero = json_dict["super_hero"] if json_dict.get("super_hero") else False
+     #Created At
+     createdAt = json_dict["created_at"] if json_dict.get("created_at") else datetime.now()
+     #super_hero = int(json_dict["super_hero"])
+     #createdAt = json_dict["created_at"]
+     #Get Record With Max Record ID in Database
+     #current_user_from_db = DBSession().query(User).order_by(User.id.desc()).first()
+     #user = User(name='Bruce Wayne1', super_hero=True, created_at=datetime.now())
+     #user = User(name=entered_name, super_hero=superHero, created_at=createdAt)
+     #DBSession.add(user)
+     # Flush to get the post.id from the database
+     #DBSession.flush()
+     #DBSession.commit()
+     with transaction.manager:
```

Comment:
Definitely. It could be a standard JSON block that includes a message (success or error) with more detail like yours above and then info on the created object referenced.  The HTTP code should always be the leading indicator of success, failure, or error cases.